### ISSUE TITLE
_get_po_paths path replacement  fails when languages are not explicitly specified

### ISSUE THREAD

**Issue by RasmusKlett** [ecometrica/django-vinaigrette#40]
When running makemessages without specifying languages (or with --all), the function _get_po_paths always returns an empty list. This results in all string hints being something like `vinaigrette.deleteme.py` instead of `app.ModelName/field:id`.

This happens because of the fix introduced in [this commit](https://github.com/ecometrica/django-vinaigrette/commit/442f36c026cd0293209257686a5cbcdaae04c4ca). Since the `locales` variable is an empty list, the `for locale in locales:` loop does nothing. 

The correct behaviour is probably to attempt to guess locale folders based on names, which as far as I can tell is what Django's makemessages does [here](https://github.com/django/django/blob/abeb4599af5ea7a384e18ec6201a95c9e419b44a/django/core/management/commands/makemessages.py#L361):
```python

        # Build locale list
        looks_like_locale = re.compile(r'[a-z]{2}')
        locale_dirs = filter(os.path.isdir, glob.glob('%s/*' % self.default_locale_path))
        all_locales = [
            lang_code for lang_code in map(os.path.basename, locale_dirs)
            if looks_like_locale.match(lang_code)
        ]

        # Account for excluded locales
        if process_all:
            locales = all_locales
        else:
            locales = locale or all_locales
            locales = set(locales).difference(exclude)
```

**Comment by RasmusKlett** [comment]
Will you respond to a pull request to fix this issue? I don't want to waste my time if not..

**Comment by jordiecometrica** [comment]
Sure, PRs always welcome. The issue doesn't affect us, so we have little motivation to fix it ourselves.

**Comment by RasmusKlett** [comment]
Well, "run the main command without args or with --all" seems like an issue everybody is bound to run into at some point. So I was at least hoping for an answer, since I've already pinpointed the problem and suggested a solution. I will take a stab at a PR, we'll see if I can find the time.

**Comment by jordiecometrica** [comment]
We'd probably be getting more complaints if everybody were running into it, especially since it's been there for so long. We always run the command with a language in mind, and I think a lot of people do too.

But like I said, we have no motivation nor reason to fix it ourselves. I don't mean to be rude, just explain why you may be waiting for a long time if you want us to fix the bug. Even with a fix identified, it takes time and effort to implement the fix, and we have no reason to do it ourselves (or mostly me, in this case, I've got other things I should be working on instead).

**Comment by RasmusKlett** [comment]
That's totally fair, I was just a little grumpy about no response for 17 days.. But I guess I forgot about  christmas break, so it is understandable.

I'm submitting a PR now. I've adapted code from the snippet I copied. I'm not quite sure what the behavior is when there are several basedirs, and they do not contain the same languages, but I hope that is such an edge case that my patch will be useful anyway
